### PR TITLE
Add support for objects created with a null prototype in objectOf

### PR DESCRIPTION
**Pull request by getkey**
It's all in the title. :-)

### CODE REVIEW

**Comment by getkey** [comment]
I would be happy to add it but isn't it a bit overkill?

**Comment by ljharb** [comment]
The maintainers may think so; but being robust is rarely overkill, especially when it's so easy.

File: factoryWithTypeCheckers.js [hunk 1]
```diff
@@ -10,6 +10,7 @@
 var emptyFunction = require('fbjs/lib/emptyFunction');
 var invariant = require('fbjs/lib/invariant');
 var warning = require('fbjs/lib/warning');
+var has = require('has');
```
Comment (gaearon) [inline factoryWithTypeCheckers.js]: Can we avoid taking on more npm deps please? Just inline it. Thanks!

**Comment by gaearon** [comment]
(Sorry for contradictory advice. Since this package is very commonly used we're trying to trim the tree a little bit.)

File: factoryWithTypeCheckers.js (hunk 1 above)
Comment (ljharb) [reply factoryWithTypeCheckers.js]: In that case, `Function.call.bind(Object.prototype.hasOwnProperty)`

**Comment by getkey** [comment]
No worries. :slightly_smiling_face:

**Comment by gaearon** [comment]
Thanks

**Pull request merged**